.github/workflows/build_and_test_cosmian_vm.yml

```diff
@@ -0,0 +1,174 @@
+    name: Test image
+    runs-on: ubuntu-22.04
+    needs: build-kms-cosmian-vm-image
+    permissions:
+      contents: 'read'
+      id-token: 'write'
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: GCP auth
+        uses: google-github-actions/auth@v2
+        with:
+          credentials_json: "${{ secrets.GOOGLE_COSMIAN_DEV_CREDENTIALS }}"
+
+      - name: Set up Google Cloud SDK
+        uses: google-github-actions/setup-gcloud@v2
+        with:
+          version: latest
+          install_components: beta
+
+      - name: Launch GCP instance
+        id: run-gcp-instance
+        env:
+          IMAGE_NAME: ${{ needs.build-kms-cosmian-vm-image.outputs.image_name }}
+          CI_INSTANCE: ${{ needs.build-kms-cosmian-vm-image.outputs.ci_instance }}
+        run: |
+          gcloud beta compute instances create $CI_INSTANCE --machine-type n2d-standard-2 \
+                                                            --zone europe-west4-a \
+                                                            --min-cpu-platform="AMD Milan" \
+                                                            --confidential-compute-type=SEV_SNP \
+                                                            --maintenance-policy=TERMINATE \
+                                                            --image="$IMAGE_NAME" \
+                                                            --image-project=$GCP_DEV_PROJECT \
+                                                            --project $GCP_DEV_PROJECT \
+                                                            --tags ssh-full,backend,http-server,https-server,cosmian-vm-agent \
+                                                            --metadata-from-file=startup-script=scripts/gcp-start.sh \
+                                                            --max-run-duration=10m \
+                                                            --instance-termination-action=DELETE
+          IP_ADDR=$(gcloud beta compute instances describe $CI_INSTANCE --format='get(networkInterfaces[0].accessConfigs[0].natIP)' --zone=europe-west4-a)
+          echo "IP_ADDR=${IP_ADDR}" >> "$GITHUB_OUTPUT"
+
+      - name: Download Cosmian VM client
+        run: |
+          wget https://package.cosmian.com/cosmian_vm/1.0.1/cosmian_vm
+
+      - name: Change permissions of binaries
+        run: |
+          chmod +x ./cosmian_vm
+
+      - name: Test Cosmian VM Agent on GCP instance
+        env:
+          IP_ADDR: ${{ steps.run-gcp-instance.outputs.IP_ADDR }}
+        run: |
+          echo "Waiting for Cosmian VM agent..."
+          until curl --insecure --output /dev/null --silent --fail https://${IP_ADDR}:5355/ima/ascii; do sleep 3; done
+          echo "[ OK ] Cosmian VM ready"
+          ./cosmian_vm --url https://${IP_ADDR}:5355 --allow-insecure-tls snapshot
+          ./cosmian_vm --url https://${IP_ADDR}:5355 --allow-insecure-tls verify --snapshot ./cosmian_vm.snapshot
```

Comment:
for my knowledge, does the 2nd arg is necessary?

Reply:
`verify` compares current snapshot from the agent, and the reference the user owns locally